Yoast/wordpress-seo
# Add settings errors message to the admin document title.

## Summary

This PR can be summarized in the following changelog entry:

* Update the document title with a message to communicate form submission errors have occurred

## Relevant technical choices:
- one of the W3C suggested techniques to inform users that errors occurred when submitting a form is by updating the document `<title>`, see https://www.w3.org/WAI/tutorials/forms/notifications/#using-the-page-title
- in particular, screen readers announce the document title as first thing when a page loads so, when a form is submitted, that is the most appropriate place for screen reader users to communicate "Hey, there are errors!"
- this PR retrieves the errors from the WordPress global `$wp_settings_errors` via `get_settings_errors()`
- this means it works wherever `add_settings_error()` is used and form submission returns errors
- for now, only the Yoast errors are taken into account
- updates the document title prepending the message `The form contains %1$s error(s).`

## Test instructions
- run `composer install`
- go to SEO > Social
- enter an invalid Twitter username
- submit
- after the page reloads, observe the document title contains the message with the error count: 1

![social](https://user-images.githubusercontent.com/1682452/61437669-efa24880-a93d-11e9-98df-b3c1f0aa25db.png)

- note: if you merge https://github.com/Yoast/bugreports/issues/457 this will work with all the other Social URL input fields as well
- go to SEO > General > Webmaster Tools
- enter invalid values in all the 4 input fields, e.g. `q w e r t y`
- submit
- after the page reloads, observe the document title contains the message with the error count: 4

![webmaster tools](https://user-images.githubusercontent.com/1682452/61437854-59baed80-a93e-11e9-902a-7fe1e8880086.png)

- note: on multisite > Network Admin > Seo > General the settings get saved with AJAX. If you want to test it there:
- disable JavaScript in your browser
- manually alter the values of the select options via your browser's dev tools
- submit

![multisite](https://user-images.githubusercontent.com/1682452/61438016-c46c2900-a93e-11e9-9150-0da97cc24350.png)

- to run the new tests, you can run only the new `input-validation` group by issuing this command: `phpunit -c phpunit-integration.xml.dist --group input-validation`

## Note
While working on this, I noticed there are a few methods to validate options that seem no longer used. Meaning that the UI has changed and doesn't allow to enter invalid values. One example of this is on multisite Network Admin > Seo > General. Those are `<select>` elements so users can only choose predefined options. Regardless, there are methods to validate and print errors if an invalid value is set. There are other cases too. What should we do with these apparently no longer used methods? /Cc @jip 

## Note 2
I noticed the License Manager and Yoast Comment Hacks add their own settings errors: should this feature be implemented also there? E.g.:

<img width="616" alt="Screenshot 2019-07-18 at 10 12 40" src="https://user-images.githubusercontent.com/1682452/61440818-b28d8480-a944-11e9-8a8b-4ce30543dbd2.png">

## Inclusion in WordPress core
I was wondering if this functionality would make more sense in WordPress. There, it could work for _all_ the settings error, not only for the Yoast ones. Any thoughts?


## UI changes

## Documentation

## Quality assurance


See https://github.com/Yoast/bugreports/issues/400
Fixes https://github.com/Yoast/bugreports/issues/466
